src/util/pointer_expr.cpp

```diff
@@ -227,13 +227,25 @@ void dereference_exprt::validate(
     "operand");
 }
 
-exprt pointer_in_range_exprt::lower() const
+exprt r_or_w_ok_exprt::lower(const namespacet &ns) const
+{
+  return conjunction(
+    {not_exprt{null_object(pointer())},
+     not_exprt{is_invalid_pointer_exprt{pointer()}},
+     not_exprt{deallocated(pointer(), ns)},
+     not_exprt{dead_object(pointer(), ns)},
```

Comment:
This condition was not present in `good_pointer_def` which `r_or_rw_ok` is now supposed to replace (cf https://github.com/diffblue/cbmc/pull/7386/files)

Reply:
I believe this was a bug in `good_pointer_def`.